### ISSUE TITLE
Allow the use of @jsweet.lang.Name with classes

### ISSUE THREAD

**Issue by negora** [cincheo/jsweet#317]
Hello:

Right now, the `@jsweet.lang.Name` annotation only can be used with:
* Packages (`ElementType.PACKAGE`)
* Methods (`ElementType.METHOD`)
* Fields (`ElementType.FIELD`)

In my humble opinion, it would be fantastic if you allowed its usage with classes too. Some JavaScript programmers use class names that start with a lowercase letter, instead of an uppercase letter. This style conflicts with the typical style rules of Java. But, with this annotation, we could rename the class definitions with proper Java names.

For example, the WNumb library uses an interface called `wNumbOptions`. I would prefer to call it `WNumbOptions` in Java but, at the same time, keep its original name when transpiling it. Something like this:

```java
@jsweet.lang.Interface
@jsweet.lang.Name (value = "wNumbOptions")
public abstract class WNumbOptions {
...
}
```
Thank you for your hard work :) .

**Comment by renaudpawlak** [comment]
Yes. That would be a nice feature but it is not that easy to implement (that why I did not do it)... Let's plan it for a not-too-far future ;)
